src/executables/master_server_monitor.hpp

```diff
@@ -26,58 +26,146 @@
 #include <cstring>
 #include <thread>
 #include <vector>
-#include <list>
+#include <queue>
+#include <unordered_map>
 #include <memory>
-#include <time.h>
+#include <mutex>
+#include <chrono>
+#include <condition_variable>
 #include "connection_defs.h"
 #include "connection_globals.h"
 
 class server_monitor
 {
+
   public:
+
+    enum class job_type
+    {
+      REGISTER_SERVER = 0,
+      UNREGISTER_SERVER = 1,
+      REVIVE_SERVER = 2,
+      CONFIRM_REVIVE_SERVER =  3,
+      JOB_MAX
+    };
+
+    server_monitor ();
+    ~server_monitor ();
+
+    server_monitor (const server_monitor &) = delete;
+    server_monitor (server_monitor &&) = delete;
+
+    server_monitor &operator = (const server_monitor &) = delete;
+    server_monitor &operator = (server_monitor &&) = delete;
+
+    void register_server_entry (int pid, const std::string &exec_path, const std::string &args,
+				const std::string &server_name,
+				std::chrono::steady_clock::time_point revive_time);
+    void remove_server_entry (const std::string &server_name);
+    void revive_server (const std::string &server_name);
+    int try_revive_server (const std::string &exec_path, std::vector<std::string> argv);
+    void check_server_revived (const std::string &server_name);
+
+    void produce_job (job_type job_type, int pid, const std::string &exec_path, const std::string &args,
+		      const std::string &server_name);
+    void consume_job ();
+
+  private:
+
+    class job
+    {
+
+      public:
+
+	job (job_type job_type, int pid, std::string exec_path, std::string args,
+	     std::string server_name);
```

Comment:
인자를 참조 형식으로 변경하면, 불필요한 생성자 호출을 막을 수 있습니다. 다음과 같이 변경 후 빌드 시 문제는 없었습니다. 참고하세요.

```suggestion
	job (job_type job_type, int pid, const std::string& exec_path, const std::string& args,
	     const std::string& server_name);
```

Reply:
수정하였습니다.